src/snowflake/cli/plugins/workspace/manager.py

```diff
@@ -0,0 +1,35 @@
+from pathlib import Path
+from typing import Dict
+
+from snowflake.cli.api.exceptions import InvalidProjectDefinitionVersion20Error
+from snowflake.cli.api.project.schemas.entities.entities import (
+    Entity,
+    v2_entity_model_to_entity_map,
+)
+from snowflake.cli.api.project.schemas.project_definition import DefinitionV20
+
+
+class WorkspaceManager:
+    def __init__(self, project_definition: DefinitionV20, project_root: Path):
+        if project_definition.definition_version != "2":
+            raise InvalidProjectDefinitionVersion20Error(
```

Comment:
nit: does it have to be this specific? InvalidProjectDefinitionVersion would do, no? We're publishing exception names for telemetry as of 2.7, let's try to contain the one-off error types.

Reply:
You're right, changed to a generic `InvalidProjectDefinitionVersionError`